src/main/java/io/confluent/copycat/jdbc/DataConverter.java

```diff
@@ -177,21 +180,20 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
       case Types.SQLXML: {
         // Some of these types will have fixed size, but we drop this from the schema conversion
         // since only fixed byte arrays can have a fixed size
-        if (nullableBuilder != null) {
-          nullableBuilder.stringType().endUnion().noDefault();
+        if (optional) {
+          builder.field(fieldName, Schema.OPTIONAL_STRING_SCHEMA);
         } else {
-          builder.type().stringType().noDefault();
+          builder.field(fieldName, Schema.STRING_SCHEMA);
         }
         break;
       }
 
       // Binary == fixed bytes
       case Types.BINARY: {
```

Comment:
Not too important, but since there's no difference between `binary` and `fixed` now, you can merge this case with the ones below it.